File: src/Overstay.Application/Features/Visas/Requests/UpdateVisaRequest.cs
namespace Overstay.Application.Features.Visas.Requests;

public class UpdateVisaRequest
{
    public DateTime ArrivalDate { get; set; }
    public DateTime ExpireDate { get; set; }
    public Guid VisaTypeId { get; set; }
    public Guid UserId { get; set; }
}

File: lib/models/update_visa_request.dart
class UpdateVisaRequest {
  final DateTime? arrivalDate;
  final DateTime? expireDate;
  final String? visaTypeId;
  final String? userId; // kun hvis du vil flytte et visa til anden bruger

  UpdateVisaRequest({
    this.arrivalDate,
    this.expireDate,
    this.visaTypeId,
    this.userId,
  });

  Map<String, dynamic> toJson() => {
    'arrivalDate': arrivalDate?.toIso8601String(),
    'expireDate': expireDate?.toIso8601String(),
    'visaTypeId': visaTypeId,
    'userId': userId,
  };
}
